src/components/navigation/quickbar.tsx

```diff
@@ -212,78 +214,84 @@ export function Quickbar({ subject, className, placeholder }: QuickbarProps) {
       </div>
     )
   }
+}
 
-  function fetchData() {
-    void fetch('https://de.serlo.org/api/stats/quickbar.json')
-      .then((res) => res.json())
-      .then((data: QuickbarData) => {
-        data.forEach((entry) => {
-          entry.pathLower = entry.path.map((x) => x.toLowerCase())
-          entry.titleLower = entry.title.toLowerCase()
-        })
+export async function fetchQuickbarData(subject?: string) {
+  const req = await fetch('https://de.serlo.org/api/stats/quickbar.json')
+  const data = (await req.json()) as QuickbarData
 
-        const subjectLower = subject?.toLowerCase() ?? ''
+  data.forEach((entry) => {
+    entry.pathLower = entry.path.map((x) => x.toLowerCase())
+    entry.titleLower = entry.title.toLowerCase()
+  })
```

Comment:
I think this'd look better being more functional. I don't like mutations in `forEach` as they are generally harder to see. Just a matter of preference I suppose.

```suggestion
  data.map((entry) => ({...entry, 
    pathLower: entry.path.map((path) => path.toLowerCase()),
    titleLower:  entry.title.toLowerCase()
  }))
```

Reply:
Yes, let's avoid mutations, they are a pain to figure out when they become bugs later 😄